crates/redis_interface/src/commands.rs

```diff
@@ -290,6 +290,46 @@ impl super::RedisConnectionPool {
             .await
     }
 
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn get_multiple_keys<K, V>(
+        &self,
+        keys: K,
+    ) -> CustomResult<Vec<Option<V>>, errors::RedisError>
+    where
+        V: FromRedis + Unpin + Send + 'static,
+        K: Into<MultipleKeys> + Send + Debug,
+    {
+        self.pool
+            .mget(keys)
+            .await
+            .into_report()
+            .change_context(errors::RedisError::GetFailed)
+    }
+
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn get_and_deserialize_multiple_keys<K, V>(
+        &self,
+        keys: K,
+        type_name: &'static str,
+    ) -> CustomResult<Vec<Option<V>>, errors::RedisError>
+    where
+        K: Into<MultipleKeys> + Send + Debug,
+        V: serde::de::DeserializeOwned,
+    {
+        let data = self.get_multiple_keys::<K, Vec<u8>>(keys).await?;
+        data.into_iter()
+            .map(|value_bytes| {
+                value_bytes
+                    .map(|bytes| {
+                        bytes
+                            .parse_struct(type_name)
```

Comment:
So this can be used only if the type of all values are same, since only one generic is passed, right? 
Let's say I query for multiple keys and pass the return value as `String` , then all those values will be deserialized to `String`.

Reply:
Yes